### DISCUSSION TITLE
trpc RESTDataSource

### DISCUSSION THREAD

**Discussion by dani-z** [trpc/trpc#2275] [Q&A]
Hi there 👋 

Thanks so so much for this amazing project! As a long graphql user I always wanted something easier to setup that would just allow me to hit the ground running. This is when I came across trpc. ❤️ 

One thing I wished trpc would have (or know how to do it) is the ability to connect my trpc resolvers to an external REST API / or SDK. I know Apollo has RESTDataSource so I was wondering if there's anything in trcp to tackle this or if there's some sort of pattern I am unaware?

Best.

**Comment by KATT** [comment]
You can use `fetch()` and just make sure your data is typed. Any sort of OpenAPI integration SDK or whatnot would work, you just need to return something typed in your resolver

**Reply by dani-z** [reply]
Thanks for your reply @KATT, appreciate all your help.
That's what I also thought but have some issues using the resolver with https://github.com/PLhery/node-twitter-api-v2
Not sure if it's to do with the fact that the SDK returns async iterators or something else 🤔

**Reply by KATT** [reply]
Yeah you can't return an async iterator in tRPC.... would be cool for the subscription API though, I've honestly never used them
